docs/content/docs/testing/errors.md

```diff
@@ -0,0 +1,290 @@
+
+The shared object registry (of type `ISharedObjectRegistry`) is something the client supplies to Fluid when the data store is loaded. From that point on, the registry is read-only. This implies the document requires certain types of DDSs for which the runtime is not able to get a factory instance. 
+
+**Steps to take**: Both the DDS's type (`channelFactoryType`) and the package path (`dataStorePackagePath`) of the data store that contains the DDS are part of the error. These indicate which DDS in a data store has the missing factory. The likely scenario is that a document was created with one configuration, then loaded with another with a newer version of the application. You’ll need to develop a migration strategy for your DDSes. 
+
+### **channelTypeNotAvailable** 
+
+This is a backwards compatibility case where an old attach message for a Distributed Data Structure (DDS) or shared object is processed. An error is thrown during the load operation when the attach message doesn't have an associated type. This DDS cannot be loaded since the runtime cannot get the factory for that DDS to create an instance of it. 
+
+**Steps to take**: You’ll need to develop a migration strategy for your DDSes to continue processing old requests. 
+
+### **Container closed without error during load** 
+
+Something closed the container while it was being loaded. This doesn’t affect your data and is not a concern. 
+
+**Steps to take**: Investigate the context in which this error is generated and make sure the closing behavior is intentional. 
+
+### **Disconnect: Summary cancelled due to summarizer or main client disconnect** 
+
+The client disconnected while the summary was being prepared. When a client disconnects, summaries should be cancelled, as there is no way to submit the summary. 
+
+**Steps to take**: This isn’t problematic unless it happens frequently on the same document. Consider more than ten occurrences to be a concern. In that case, report this issue to the Fluid team by filing a [GitHub issue](https://github.com/microsoft/FluidFramework/issues). 
+
+### **DuplicateJoinSessionRefresh** 
+
+The client refreshed or joined the session twice. 
+
+**Steps to take**: This is a known issue in the Fluid Framework. It is being investigated and this article will be updated when a fix is deployed. 
+
+### **Epoch mismatch** 
+
+The client and server epochs are different. The epoch can change when a previous version of document is restored. If there is already a client running with another epoch when this happens, further interaction with storage leads to this error. 
+
+**Steps to take**: Refreshing the document should fetch the new epoch and cause the document to load properly. 
+
+### **Error parsing snapshot response: 0x3dc** 
+
+The client was unable to parse a fetch snapshot response from the service. 
```

Comment:
the phrase "a fetch snapshot response" is puzzling. And "fetch" which is normally a verb is being used as an adjective. Can this sentence be reworded?